### PR TITLE
Adding more dataframe example to read csv files

### PR DESCRIPTION
**Pull request by DataPsycho**
# Which issue does this PR close?

Not related to pr but to improve document

Closes #.

# Rationale for this change

Did not proposed any change just added example

# What changes are included in this PR?

change in the dataframe example file

# Are these changes tested?

Current implementation does not have any test to run

# Are there any user-facing changes?

No

No Breaking change in public API.

### CODE REVIEW

File: datafusion-examples/examples/dataframe.rs [hunk 1]
```diff
@@ -41,3 +44,37 @@ async fn main() -> Result<()> {
 
     Ok(())
 }
+
+// Example to read data from a csv file with inferred schema
+async fn example_read_csv_file_with_inferred_schema() -> Arc<DataFrame> {
+    let path = "example.csv";
+    // Create the data to put into the csv file with headers
+    let content = "id,time,vote,unixtime,rating\na1,\"10 6, 2013\",3,1381017600,5.0\na2,\"08 9, 2013\",2,1376006400,4.5";
```
Comment (martin-g) [inline datafusion-examples/examples/dataframe.rs:52]:
```suggestion
    let content = r#"id,time,vote,unixtime,rating
    a1,\"10 6, 2013\",3,1381017600,5.0
    a2,\"08 9, 2013\",2,1376006400,4.5"#;
```

File: datafusion-examples/examples/dataframe.rs [hunk 2]
```diff
@@ -41,3 +44,37 @@ async fn main() -> Result<()> {
...
+    // write the data
+    fs::write(path, content).expect("Problem with writing file!");
+    // Create a session context and create a lazy
```
Comment (martin-g) [inline datafusion-examples/examples/dataframe.rs:55]: `... create a lazy ???`
the sentence seems unfinished

File: datafusion-examples/examples/dataframe.rs [hunk 3]
```diff
@@ -41,3 +44,37 @@ async fn main() -> Result<()> {
...
+    let ctx = SessionContext::new();
+    let df = ctx.read_csv(path, CsvReadOptions::default()).await.unwrap();
+    df
+}
+
+// Example to read csv file with a given csv file
+async fn example_read_csv_file_with_schema() -> Arc<DataFrame> {
+    let path = "example.csv";
+    // Create the data to put into the csv file with headers
+    let content = "id,time,vote,unixtime,rating\na1,\"10 6, 2013\",3,1381017600,5.0\na2,\"08 9, 2013\",2,1376006400,4.5";
```
Comment (martin-g) [inline datafusion-examples/examples/dataframe.rs:65]:
```suggestion
    let content = r#"id,time,vote,unixtime,rating
    a1,\"10 6, 2013\",3,1381017600,5.0
    a2,\"08 9, 2013\",2,1376006400,4.5"#;
```

File: datafusion-examples/examples/dataframe.rs (hunk 2 above)
Comment (DataPsycho) [reply datafusion-examples/examples/dataframe.rs]: Updated as suggested. and Completed the incomplete instruction

File: datafusion-examples/examples/dataframe.rs [hunk 4]
```diff
@@ -41,3 +44,47 @@ async fn main() -> Result<()> {
...
+        Field::new("id", DataType::Utf8, false),
+        Field::new("time", DataType::Utf8, false),
+        Field::new("vote", DataType::Int32, true),
+        Field::new("unixtime", DataType::Int64, false),
+        Field::new("rating", DataType::Float32, true),
+    ]);
+    // Create a csv option provider
+    let mut csv_read_option = CsvReadOptions::default();
+    // Update the option provider with the defined schema
+    csv_read_option.schema = Some(&schema);
```
Comment (alamb) [inline datafusion-examples/examples/dataframe.rs:86]: If you wanted to use a slightly more idiomatic rust syntax (and avoid `mut`), you could do:

```suggestion
    // Create a csv option provider with the desired schema 
    let csv_read_option = CsvReadOptions {
      // Update the option provider with the defined schema
      schema: Some(&schema),
      ..default::Default()
    };
```

**Review by alamb — APPROVED** [review]
Thank you @DataPsycho

File: datafusion-examples/examples/dataframe.rs (hunk 4 above)
Comment (DataPsycho) [reply datafusion-examples/examples/dataframe.rs]: Committed, suggested Changes.

**Review by andygrove — APPROVED** [review]
Thanks @DataPsycho. I appreciate the effort to add more docs/examples to help new users.

File: datafusion-examples/examples/dataframe.rs [hunk 5]
```diff
@@ -41,3 +44,66 @@ async fn main() -> Result<()> {
...
+
+// Example to read data from a csv file with inferred schema
+async fn example_read_csv_file_with_inferred_schema() -> Arc<DataFrame> {
+    let path = "example.csv";
+    // Create the data to put into the csv file with headers
+    let content = r#"id,time,vote,unixtime,rating
+    a1,\"10 6, 2013\",3,1381017600,5.0
+    a2,\"08 9, 2013\",2,1376006400,4.5"#;
+    // write the data
+    fs::write(path, content).expect("Problem with writing file!");
```
Comment (andygrove) [inline datafusion-examples/examples/dataframe.rs:56]: minor nit: this code is repeated a few times and could potentially be moved to its own function

Comment (DataPsycho) [reply datafusion-examples/examples/dataframe.rs:56]: Sure thing. As my first commit, I just want to add some stuff. I have a lot more examples to add then I will update it and create a separate function for each type of data files like excel, json etc.

Comment (DataPsycho) [reply datafusion-examples/examples/dataframe.rs]: I have refactored it according to your suggestion. Thanks

**Comment by alamb** [comment]
Thanks again @DataPsycho !

**Pull request merged**